packages/ui-extensions-dev-console/src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,22 @@
+import styles from './Icon.module.scss'
+import {VisuallyHidden} from '../VisuallyHidden/VisuallyHidden.js'
+import React from 'react'
+
+export interface IconProps {
+  /** The SVG contents to display in the icon (icons should fit in a 20 × 20 pixel viewBox) */
+  source: React.FunctionComponent<React.SVGProps<SVGSVGElement>> | string
+
+  /** Descriptive text to be read to screenreaders */
+  accessibilityLabel: string
+}
+
+export function Icon({source, accessibilityLabel}: IconProps) {
+  const SourceComponent = source
+
+  return (
+    <span className={styles.Icon}>
+      <VisuallyHidden>{accessibilityLabel}</VisuallyHidden>
+      <SourceComponent className={styles.Svg} focusable="false" aria-hidden="true" />
```

Comment:
Would this work if `source` is a string?

Reply:
Not right now. I simplified the use case assuming that we wouldn't need it and we can add it back easily if we do.  Thoughts?